ArduPlane/altitude.cpp

```diff
@@ -358,26 +358,37 @@ int32_t Plane::calc_altitude_error_cm(void)
 }
 
 /*
-  check for FBWB_min_altitude_cm violation
- */
-void Plane::check_fbwb_minimum_altitude(void)
+  check for min and max altitude: FBWB_min_altitude_cm and fbwb_max_altitude_m violation
+*/
+void Plane::check_fbwb_altitude(void)
 {
-    if (g.FBWB_min_altitude_cm == 0) {
+    float max_alt_cm = fence.get_safe_alt_max()*100.0f;
+    bool should_check_max = fence.enabled() && (max_alt_cm > 0);
+    bool should_check_min = g.FBWB_min_altitude_cm != 0;
```

Comment:
we'll also need to use the FENCE_MARGIN, so we stay below the margin of the fence alt max

Reply:
@tridge get_safe_alt_max is giving  **alt_max - margin**. I wonder if there is any mistake, please correct me.